File: app/src/main/java/org/dhis2/usescases/datasets/datasetInitial/DataSetInitialModule.java
package org.dhis2.usescases.datasets.datasetInitial;

import org.dhis2.data.dagger.PerActivity;
import org.dhis2.data.schedulers.SchedulerProvider;
import org.hisp.dhis.android.core.D2;

import dagger.Module;
import dagger.Provides;

/**
 * QUADRAM. Created by ppajuelo on 24/09/2018.
 */
@PerActivity
@Module
public class DataSetInitialModule {
    private final String dataSetUid;
    private final DataSetInitialContract.View view;

    DataSetInitialModule(DataSetInitialContract.View view, String dataSetUid) {
        this.view = view;
        this.dataSetUid = dataSetUid;
    }

    @Provides
    @PerActivity
    DataSetInitialContract.View provideView(DataSetInitialActivity activity) {
        return activity;
    }

    @Provides
    @PerActivity
    DataSetInitialContract.Presenter providesPresenter(DataSetInitialRepository dataSetInitialRepository, SchedulerProvider schedulerProvider) {
        return new DataSetInitialPresenter(view, dataSetInitialRepository, schedulerProvider);
    }

    @Provides
    @PerActivity
    DataSetInitialRepository dataSetInitialRepository(D2 d2) {
        return new DataSetInitialRepositoryImpl(d2, dataSetUid);
    }
}

File: app/src/main/java/org/dhis2/usescases/datasets/datasetInitial/DataSetInitialModule.kt
package org.dhis2.usescases.datasets.datasetInitial

import dagger.Module
import dagger.Provides
import org.dhis2.data.dagger.PerActivity
import org.dhis2.data.schedulers.SchedulerProvider
import org.hisp.dhis.android.core.D2

/**
 * QUADRAM. Created by ppajuelo on 24/09/2018.
 */
@PerActivity
@Module
class DataSetInitialModule internal constructor(
    private val view: DataSetInitialView,
    private val dataSetUid: String
) {

    @Provides
    @PerActivity
    internal fun provideView(activity: DataSetInitialActivity): DataSetInitialView {
        return activity
    }

    @Provides
    @PerActivity
    internal fun providesPresenter(
        dataSetInitialRepository: DataSetInitialRepository,
        schedulerProvider: SchedulerProvider
    ): DataSetInitialPresenter {
        return DataSetInitialPresenter(view, dataSetInitialRepository, schedulerProvider)
    }

    @Provides
    @PerActivity
    internal fun dataSetInitialRepository(d2: D2): DataSetInitialRepository {
        return DataSetInitialRepositoryImpl(d2, dataSetUid)
    }
}
